src/components/frame/Extensions/pages/ExtensionDetailsPage/partials/ExtensionDetailsSidePanel/UninstallButton.tsx

```diff
@@ -133,16 +139,15 @@ const UninstallButton = ({
   const [isCheckboxChecked, setIsCheckboxChecked] = useState(false);
   const { handleUninstall, isLoading } = useUninstall(extensionId);
   const { waitingForActionConfirmation } = useExtensionDetailsPageContext();
+  const { hasActiveStream, hasUnclaimedFunds } =
+    useGetActiveAndUnclaimedStreams();
 
   const isStreamingPaymentsExtension =
     extensionId === Extension.StreamingPayments;
 
-  // @todo: add proper logic here to determine if there are active streams or unclaimed funds
-  const hasActiveStream = false;
-  const hasUnclaimedFunds = false;
-
-  const shouldShowWarning =
-    isStreamingPaymentsExtension && (hasActiveStream || hasUnclaimedFunds);
+  const canRemoveExtension =
+    !isStreamingPaymentsExtension ||
+    (isStreamingPaymentsExtension && (!hasActiveStream || !hasUnclaimedFunds));
```

Comment:
I believe this condition should be `isStreamingPaymentsExtension && !hasActiveStream && !hasUnclaimedFunds` as the extension can be removed if there are no active streams and no unclaimed funds. If one of them (hasActiveStream or hasUnclaimedFunds) is true, we should block the extension uninstall